### ISSUE TITLE
Documentation on actions in ops missing information on how to get info out from actions

### ISSUE THREAD

**Issue by erik78se** [canonical/jaas-dashboard#1199]
I'm reading the docs to try figuring out how to get the "message" field in the GUI to show the result from my actions.

But its nothing there that helps.

What should I do to get anything displayed there?

This is the docs I'm referring to: https://operator-framework.readthedocs.io/en/latest/_modules/ops/charm.html?highlight=set_results#

**Comment by erik78se** [comment]
https://github.com/canonical-web-and-design/jaas-dashboard/issues/1035

**Comment by erik78se** [comment]
Is this where this thing goes?

https://github.com/canonical-web-and-design/jaas-dashboard/blob/ddf52fc4de84dfc0f658fcb89ca01fd933e9dec3/src/pages/EntityDetails/Model/ActionLogs/ActionLogs.tsx#L200

How do I connect this code from my charm in such a way that the "message" field turns up there? 

```
def _on_get_session_key_action(self, event):
	event.log("Getting new session key through rpc...")
	key = rpc.get_session_key()
	if key:
            event.set_results(results={'session-key': key})
        else:
            event.fail("Unable to get new session key")
```

**Comment by erik78se** [comment]
Is there any chance to get this feature in the pipe?

Can I help in any way?

It is painful to have all these good actions, but no way to interact with them in the GUI. This would help massively to operators and we would use it alot.

**Comment by anthonydillon** [comment]
Hi @erik78se this is high on our priority list. I will bump this into investigation/development stage now and a member of the team will be ready to work on it shortly.

**Comment by anthonydillon** [comment]
@Lukewh not sure which epic this should go in

**Comment by erik78se** [comment]
I hope you can get this feature right. It will be a game changer for many reasons.

**Comment by erik78se** [comment]
WOW, when will this go into a release?

Den tors 6 okt. 2022 23:02Goulin Khoge ***@***.***> skrev:

**Comment by goulinkh** [comment]
Hey Erik, we are still in the process of updating old dependencies in the dashboard but as far as I know, the new dashboard will be shipped with Juju 3.0 (which we very recently added support for).

However if would like to try it, here is the [latest version](https://drive.google.com/file/d/1bbyvE-k2oYYLbGOi_ROMneJz7NZEZO7J/view?usp=sharing) of the dashboard that works with Juju version 2.9 and 3.0 beta4.

**Comment by erik78se** [comment]
> However if would like to try it, here is the [latest version](https://drive.google.com/file/d/1bbyvE-k2oYYLbGOi_ROMneJz7NZEZO7J/view?usp=sharing) of the dashboard that works with Juju version 2.9 and 3.0 beta4.

I tried it, but it doesn't seem to work.

I ran the action from the GUI which gave the action id = 25. But nothing of the output shows in the GUI.

![image](https://user-images.githubusercontent.com/850603/195145407-93a2d71a-1ba3-43e8-8b05-ace41d4d576e.png)

I can get the output from the terminal as expected.

![image](https://user-images.githubusercontent.com/850603/195145281-0d670d5b-2447-46a3-9540-7e9ab62cd10d.png)

Another observation is that the "Output" "STDOUT/STDERR" dropdown, doesn't work either.

![image](https://user-images.githubusercontent.com/850603/195146064-571177f6-3bb1-440e-b9e2-82dc051f35ec.png)
